### DISCUSSION TITLE
Autodetecting filetype of compressed file

### DISCUSSION THREAD

**Discussion by avidseeker** [neovim/neovim#31300] [Q&A]
In Vim, `:set ft` returns the expected `markdown`
```
$ echo test | xz -c > test.md.xz && vim --clean test.md.xz
```
In Neovim, `nvim --clean test.md.xz`, the `:set ft` is not set.

What's the option for getting the default Vim behavior? Or is this a bug?

**Comment by zeertzjq** [comment]
It seems to be a regression from #27931

**Reply by zeertzjq** [reply]
Ah, it seems that autocmd pattern matching actually matches with both the unexpanded filename and the expanded file name. This is complicated...

**Reply by zeertzjq** [reply]
Created an issue #31306
